src/interface/emacs/khoj.el

```diff
@@ -437,43 +474,53 @@ Auto invokes setup steps on calling main entrypoint."
                                          (if content-type (format "%s " content-type) "all")
                                          (string-trim (format "%s %s" (nth 1 (nth 1 status)) (nth 2 (nth 1 status))))
                                          (if (> (- (point-max) (point)) 0) (format ". Response: %s" (string-trim (buffer-substring-no-properties (point) (point-max)))) ""))))))
-                        nil t t)))
-    (setq khoj--indexed-files files-to-index)))
+                        nil t t))))
 
-(defun khoj--render-files-as-request-body (files-to-index previously-indexed-files boundary)
+(defun khoj--render-update-files-as-request-body (files-to-index boundary)
   "Render `FILES-TO-INDEX', `PREVIOUSLY-INDEXED-FILES' as multi-part form body.
 Use `BOUNDARY' to separate files. This is sent to Khoj server as a POST request."
   (with-temp-buffer
     (set-buffer-multibyte nil)
     (insert "\n")
     (dolist (file-to-index files-to-index)
       ;; find file content-type. Choose from org, markdown, pdf, plaintext
-      (let ((content-type (cond ((string-match "\\.org$" file-to-index) "text/org")
-                                ((string-match "\\.\\(md\\|markdown\\)$" file-to-index) "text/markdown")
-                                ((string-match "\\.pdf$" file-to-index) "application/pdf")
-                                (t "text/plain"))))
+      (let ((content-type (khoj--filename-to-mime-type file-to-index))
+            (file-name (encode-coding-string  file-to-index 'utf-8)))
       (insert (format "--%s\r\n" boundary))
-      (insert (format "Content-Disposition: form-data; name=\"files\"; filename=\"%s\"\r\n" file-to-index))
+      (insert (format "Content-Disposition: form-data; name=\"files\"; filename=\"%s\"\r\n" file-name))
       (insert (format "Content-Type: %s\r\n\r\n" content-type))
       (insert (with-temp-buffer
                 (insert-file-contents-literally file-to-index)
                 (buffer-string)))
       (insert "\r\n")))
-    (dolist (file-to-index previously-indexed-files)
-      (when (not (member file-to-index files-to-index))
-        ;; find file content-type. Choose from org, markdown, pdf, plaintext
-        (let ((content-type (cond ((string-match "\\.org$" file-to-index) "text/org")
-                                  ((string-match "\\.\\(md\\|markdown\\)$" file-to-index) "text/markdown")
-                                  ((string-match "\\.pdf$" file-to-index) "application/pdf")
-                                  (t "text/plain"))))
+    (insert (format "--%s--\r\n" boundary))
+    (buffer-string)))
+
+(defun khoj--render-delete-files-as-request-body (delete-files boundary)
+  "Render `DELETE-FILES' as multi-part form body.
+Use `BOUNDARY' to separate files. This is sent to Khoj server as a POST request."
+  (with-temp-buffer
+    (set-buffer-multibyte nil)
+    (insert "\n")
+    (debug delete-files)
```

Comment:
Should we remove this debug statement `debug delete-files` if it's only used during development?
```suggestion
```

Reply:
Forgot to delete the test code, sorry